var/ramble/repos/builtin/applications/openfoam-org/application.py

```diff
@@ -0,0 +1,247 @@
+
+    executable('patchSummary', 'patchSummary -parallel', use_mpi=True,
+               redirect='{experiment_run_dir}/log.patchSummary')
+    executable('checkMesh', "checkMesh -parallel -constant", use_mpi=True,
+               redirect='{experiment_run_dir}/log.checkMesh')
+    executable('potentialFoam', 'potentialFoam -parallel {potential_flags}', use_mpi=True,
+               redirect='{experiment_run_dir}/log.potentialFoam')
+    executable('simpleFoam', 'simpleFoam -parallel {simple_flags}', use_mpi=True,
+               redirect='{experiment_run_dir}/log.simpleFoam')
+
+    workload_variable('export_prefix', default='-x',
+                      description='Prefix for exporting an environment variable with mpirun',
+                      workloads=['*'])
+
+    workload_variable('workload_exports', default='',
+                      description='Placeholder variable which holds all variable exports. Defined during setup.',
+                      workloads=['*'])
+
+    workload_variable('export_variables',
+                      description='Comma separated list of all env-var names that need to be exported',
+                      default="PATH,LD_LIBRARY_PATH,FOAM_APP,FOAM_APPBIN,FOAM_ETC," +
+                              "FOAM_EXT_LIBBIN,FOAM_INST_DIR,FOAM_JOB_DIR,FOAM_LIBBIN," +
+                              "FOAM_MPI,FOAM_RUN,FOAM_SETTINGS,FOAM_SIGFPE,FOAM_SITE_APPBIN," +
+                              "FOAM_SITE_LIBBIN,FOAM_SOLVERS,FOAM_SRC,FOAM_TUTORIALS," +
+                              "FOAM_USER_APPBIN,FOAM_USER_LIBBIN,FOAM_UTILITIES,WM_ARCH," +
+                              "WM_ARCH_OPTION,WM_CC,WM_CFLAGS,WM_COMPILER,WM_COMPILER_LIB_ARCH," +
+                              "WM_COMPILER_TYPE,WM_COMPILE_OPTION,WM_CXX,WM_CXXFLAGS,WM_DIR," +
+                              "WM_LABEL_OPTION,WM_LABEL_SIZE,WM_LDFLAGS,WM_LINK_LANGUAGE,WM_MPLIB," +
+                              "WM_OPTIONS,WM_OSTYPE,WM_PRECISION_OPTION,WM_PROJECT,WM_PROJECT_DIR," +
+                              "WM_PROJECT_INST_DIR,WM_PROJECT_USER_DIR,WM_PROJECT_VERSION," +
+                              "WM_THIRD_PARTY_DIR,MPI_ARCH_FLAGS,MPI_ARCH_INC,MPI_ARCH_LIBS," +
+                              "MPI_ARCH_PATH,MPI_BUFFER_SIZE,MPI_ROOT",
+                      workloads=['*'])
+
+    log_prefix = os.path.join(Expander.expansion_str('experiment_run_dir'), 'log.')
+
+    figure_of_merit('Number of cells', log_file=(log_prefix + 'snappyHexMesh'),
+                    fom_regex=r'Layer mesh\s+:\s+cells:(?P<ncells>[0-9]+)\s+.*',
+                    group_name='ncells', units='')
+
+    figure_of_merit('snappyHexMesh Time ({n_ranks_hex} ranks)', log_file=(log_prefix + 'snappyHexMesh'),
+                    fom_regex=r'Finished meshing in = (?P<mesh_time>[0-9]+\.?[0-9]*).*',
+                    group_name='mesh_time', units='s')
+
+    figure_of_merit('simpleFoam Time ({n_ranks} ranks)', log_file=(log_prefix + 'simpleFoam'),
```

Comment:
Are we sure we want a dynamic fom name based on `n_ranks`? Stuffing values in the keys/LHS like this makes comparing across runs much harder

I think it should probably be 2 FOMs:

1. Ranks used for each bit
2. Time for each bit 

That way we still support `SELECT * from FOMs where fom_name == 'simpleFoam Time'`. That is much harder when it's dynamic

Reply:
OK sure, so you want `simpleFoam ranks` as a separate FOM, right? That's easy to add. I really just wanted to make it clear in the FOMs that the number of ranks between the parts are allowed to be different.

Reply:
Yea I think it's kind of tricky. I agree it's important that we clearly mark that, but I think having the key/LHS change is kind of dangerous

In my mind different runs change overall `n_ranks` and we don't mark that in the FOM (so we can better compare) so it's better to follow that pattern here

I think a FOM tracking how many of the ranks used for the decomposition makes most sense, but I'm open to other ideas

Reply:
OK. I can add this. I think we might want to make an internal task for converting a variable into a FOM directly, as that isn't possible right now. So, something gross has to happen (like just writing the variable out into a file so we can then extract it again as a FOM). I just pushed an update though. Let me know what you think.